Editions on Competencies.m (#199)
CaptainSifff/paper_teaching-learning-RSE: competencies.md

"The Missing Semester of Your CS Education" [@MIT], which aims to convey computing
ecosystem literacy even to students of Computer Science at MIT.

Researchers investing increasing amounts of their time developing their software engineering (SE)
skills to support their research work can find themselves with little time to do the research
itself.
This, in turn, presents career development challenges as the experience required to 
progress in research and academic roles is traditionally assessed through metrics that
do not directly include software outputs.
A recent shift towards the establishment of the distinct role of a
_"Research Software Engineer"_ [@WhatResearchSoftware]
(RSE, a term that emerged from the United Kingdom a little over 10 years ago [@Hettrick2016]),
now provides a base on which sustainable career opportunities can be (and are being) built,
allowing for better training of researchers and more effective support for the development of high
quality research software, and therefore higher quality research.
There is still a long way to go, but positive change is well underway.

> RSEs may work within one of the increasing number of research SE teams that

Author: In that case, RSE would be the way to go. Fixed already

the training and continuous professional development of RSEs,
thus helping to provide a positive impact on
research outputs and, ultimately, society as a whole.
These competencies draw upon skills from traditional SE practice,
established research culture, and the commitment to being part of a team.
While there is anecdotal evidence that academic SE practitioners
would have chosen different competencies, we will argue that this trifecta of skills
is exactly what is needed for modern digital research.

> While this paper is based on workshop discussions that were attended largely by Research Software Engineers in Germany (deRSE23 in Paderborn, un-deRSE23 in Jena),

Author: I had understood that the discussions for this paper in particular came from RSEs in Germany (that happened to meet at those conferences). Who could confirm what the intended meaning is?

to use all the skills highlighted to the same level of expertise.
Therefore, @sec:how-much-to-know examines how much a person
needs to know depending on their education or career level
or the type of projects they would like to be involved with.
In the same section, we provide an overview of what skills and limitations
an RSE in different team structures typically has, and we give

> recommendations for organisations to support RSEs.

Reviewer: I was struggling with this formulation before, but I think that now the meaning of the sentence changes:

- Before: The organisations need to support RSEs, we tell them how they can do that
- Now: We recommend the organisations to supports RSEs -> this is already clear to them

Author: But any organisation having RSEs would 'need' to support them. It would be odd having a group of employees that do not need to be supported. I would read it as the recommendations are on how organisations can effectively support RSEs taking into account consideration the different team structures.
The full sentence is 
> In the same section, we provide an overview of what skills and limitations an RSE in different team structures typically has, and we give recommendations for organisations to support them.
Maybe we can change it to
> In the same section, we provide an overview of what skills and limitations an RSE in different team structures typically has, and we give recommendations for organisations to support them accordingly.

followed by a formulation of a plan to fulfil them that is finally implemented.
This is then followed by testing that things work as expected and that they continue to do so into the future.
Often the development is iterated.

> We emphasise that the life-cycle is not a close-end process as it also includes periods of maintaining a software

Reviewer: ```suggestion
We emphasise that the life-cycle is not a close-end process, as it also includes periods of maintaining a software
```

and also withdrawing software from its original use.
The RSE should be aware of this life-cycle
and be able to predict and cater to the changing needs of software as it moves through the stages.

### Use repositories (SWREPOS)

RSEs should be able to identify and use fitting public platforms (so-called software repositories or repos) to share the artefacts they have
created and invite the public to scrutinise them for public review. 

> A good practice supported by repositories and also aligned to the software life-cycle is the use of releases as they provide a frozen version of a software, 

Reviewer: ```suggestion
A good practice supported by repositories and also aligned to the software life-cycle is the use of releases, as they provide a frozen version of a software, 
```

corresponding to a set of requirements and functionalities. 

> As the software grows to include new features, new version releases will come making it easier for users (including RSEs using building upon) to catch up with changes.

Reviewer: This is new, I think it nicely fits here.

> As the software grows to include new features, new version releases will come making it easier for users (including RSEs using building upon) to catch up with changes.

Reviewer: This is new, I think it nicely fits here.
Maybe it needs a second opinion, since it is new content.
FYI @CaptainSifff

Finally, RSEs will need to properly execute the technicalities of software publishing,
such as applying licences, honouring copyright statements and crediting contributors.

> ### Using domain repositories/registries/directories (DOMREP)

Reviewer: I would argue that, if the heading is tending to get longer, let's just restrict it to one term:

`Using domain repositories (DOMREP)` (GitHub suggestion does not seem to work here)

It is anyway specified in the paragraph.

Reviewer 2: yes, I like this variant better, since it maps to the abbreviation. Depending on hiow mauch the text lengthens...

Almost all research software is developed within a specific scientific domain.
Some software may be able to cross boundaries, but the majority will have a
home domain, with which it needs to be able to interact. The RSE needs to be

> aware of any domain specific software repositories, data sets and catalogues (aka as registries or directories).

Reviewer: Is the "aka" maybe too colloquial? In any case, it includes the `as`.

```suggestion
aware of any domain specific software repositories, data sets and catalogues (also known as registries or directories).
```

The RSE also needs to be aware of how their software can interact with the existing
domain-specific data repositories.
While software repositories (see SWREPOS in the previous section) are commonly used to actually host the software and follow its development, 
domain-based repositories/registries/directories provide a quick description of the software together with some links to other research outcomes. 
Such descriptions and links can be represented with metadata (improving FAIRness of the software) and be exposed to the broad public, 

> making it easier for others (people and machines) to find, for instance, software used of a partiucular task, taking a specific dataset as imput, dealing with a certain type of data.

Reviewer: I think this is what you mean:

```suggestion
making it easier for others (people and machines) to find, for instance, software used for a partiucular task, taking a specific dataset as imput, or dealing with a certain type of data.
```

> In addition to software repositores and domain-based repositories, archives are also used in research for (software) long-term preservation. 

Reviewer: New content (FYI @CaptainSifff).

These skills, while already numerous are also generic on purpose. They span a
multidimensional space in which the day-to-day tasks and responsibilities of an
RSE can be found. A snapshot of what this means today was obtained
from learners and novice RSEs that we surveyed during the Paderborn deRSE workshop.

> The question was what they would have like to learn before becoming an RSE. Among the top five things mentioned were:

Reviewer: ```suggestion
The question was what they would have liked to learn before becoming an RSE. Among the top five things mentioned were:
```

  since the contributed code has to be understood by others. Interacting with
  project members depends on the TEAM skill. Today this entails the effective
  use of collaborative platforms like GitHub/GitLab, honouring a project's code of

>   conduct, and some knowledge of popular software licences like e.g., the General Public License (GPL).

Reviewer: In this paper, I see everywhere `e.g.` used without a comma after it (but one before).

```suggestion
  conduct, and some knowledge of popular software licences like, e.g. the General Public License (GPL).
```

Several instances of this below. I also prefer the version with the comma after it, but it is a matter of chosen style: https://english.stackexchange.com/a/16215/219523

Author: I saw both, e.g. and e.g., so I picked one. Either way, we should use only one style. What is the chosen one?

  research software - often a researcher with RSE skills. Often time-constrained,
      may be self-taught.
- **Individual developer (RSE team-based):** A single person working on research
  software - generally a professional RSE assigned to support another team's software on their own,

>   who however is connected to an RSE team.

Author: Ah, that is the meaning of "who however is connected to an RSE team" I will reverse

In the table above, we have looked at how different competencies can be related
to and handled by researchers and RSEs working in different environments within
an organisation and how the organisations themselves can contribute.
We recognise that this is challenging getting a detailed view 
and that our content in the table is still a generalisation.
We talk about the "research software engineer" as a single entity but as the field expands,
we expect to see more roles and job titles emerging around the RSE concept,
many of which fit under the wider umbrella of research technology professionals (RTPs).
Examples might differ from other RSE-like computational roles, 
e.g., EMBL-EBI's BioExcel competency framework [@BIOEXCEL] (also @subsec:emblbio),

> as ours include a range of different roles (mostly) taken from King's Digital Lab at King's College London [@KDL].

Author: I see now what was the intended meaning, but that is not what was written. "is a range" was what told me that the examples were different from EMBL-EBI but taken from KCL. I will rewrite.

# RSE specialisations {#sec:rse-specialisations}

> What we have defined above are intended to be base skills that an RSE —irrespective of domain, place, and time, should know about.

Author: Is there a space after a long dash?
